presto-docs/src/main/sphinx/functions/aggregate.rst

```diff
@@ -113,7 +113,30 @@ General Aggregate Functions
     for each input value. In addition to taking the input value, ``inputFunction``
     takes the current state, initially ``initialState``, and returns the new state.
     ``combineFunction`` will be invoked to combine two states into a new state.
-    The final state is returned::
+    The final state is returned. Throws an error if ``initialState`` is NULL.
+    The behavior is undefined if ``inputFunction`` or ``combineFunction`` return a NULL.
+
+    Take care when designing ``initialState``, ``inputFunction`` and ``combineFunction``.
+    These need to support evaluating aggregation in a distributed manner using partial
+    aggregation on many nodes, followed by shuffle over group-by keys, followed by
+    final aggregation. Given a set of all possible values of state, make sure that
+    combineFunction is `commutative <https://en.wikipedia.org/wiki/Commutative_property>`_
+    and `associative <https://en.wikipedia.org/wiki/Associative_property>`_
+    operation with initialState as the
+    `identity <https://en.wikipedia.org/wiki/Identity_element>`_ value.
```

Comment:
Did you locally generate these docs to ensure they are properly formatted?  I think we're missing `::` here to ensure the below is formatted as code.

Reply:
I didn't think of formatting it as code. Just copy-pasted from Velox:

https://facebookincubator.github.io/velox/functions/presto/aggregate.html#reduce_agg

Reply:
OK seems fine, thanks!